scipy/stats/_AB_SW.pyx

```diff
@@ -0,0 +1,501 @@
+# cython: language_level=3
+# cython: boundscheck=False
+# cython: initializedcheck=False
+# cython: nonecheck=False
+# cython: wraparound=False
+# cython: cdivision=True
+# cython: cpow=True
+
+from libc.math cimport exp, sqrt, abs, log, acos
+import numpy as np
+cimport numpy as cnp
+cnp.import_array()
+
+
+def gscale(int test, int other):
+    """
+    Cython translation for the FORTRAN 77 code given in:
+
+    Dinneen, L. C. and Blakesley, B. C., "Algorithm AS 93: A Generator for the
+    Null Distribution of the Ansari-Bradley W Statistic", Applied Statistics,
+    25(1), 1976, :doi:`10.2307/2346534`
+    """
+    cdef int m = min(test, other), n = max(test, other)
+    cdef int astart = ((test + 1) // 2) * (1 + (test // 2))
+    cdef int LL = (test * other) // 2 + 1
```

Comment:
Unfortunately, since this is an exact test, it starts overflowing very quickly around `m=n=150`. Hence it won't reach critical  values to consider that.

Reply:
Okay, sounds good. Resolving.